src/v/cloud_storage/remote_segment.cc

```diff
@@ -556,7 +563,24 @@ ss::future<> remote_segment::run_hydrate_bg() {
                 // cache can't delete it until we close it.
                 auto tx_path = generate_remote_tx_path(_path);
                 auto segment_status = co_await _cache.is_cached(_path);
+                segment_was_cached |= segment_status
+                                      != cache_element_status::not_available;
                 auto txrange_status = co_await _cache.is_cached(tx_path);
+                txrange_was_cached |= txrange_status
+                                      != cache_element_status::not_available;
+
+                if (
+                  (txrange_status == cache_element_status::not_available
+                   && txrange_was_cached)
+                  || (segment_status == cache_element_status::not_available && segment_was_cached)) {
```

Comment:
Maybe `txrange_was_cached` and `segment_was_cached` are supposed to be updated after the condition? I don't really understand how this supposed to work otherwise.